superset/utils/webdriver.py

```diff
@@ -0,0 +1,132 @@
+from selenium.webdriver.common.by import By
+from selenium.webdriver.remote.webdriver import WebDriver
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.ui import WebDriverWait
+
+from superset.extensions import machine_auth_provider_factory
+from superset.utils.urls import headless_url
+
+WindowSize = Tuple[int, int]
+logger = logging.getLogger(__name__)
+
+# Time in seconds, we will wait for the page to load and render
+SELENIUM_CHECK_INTERVAL = 2
+SELENIUM_RETRIES = 5
+SELENIUM_HEADSTART = 3
+
+
+if TYPE_CHECKING:
+    # pylint: disable=unused-import
+    from flask_appbuilder.security.sqla.models import User
+
+
+class WebDriverProxy:
+    def __init__(
+        self, driver_type: str, window: Optional[WindowSize] = None,
+    ):
+        self._driver_type = driver_type
+        self._window: WindowSize = window or (800, 600)
+
+    def create(self) -> WebDriver:
+        if self._driver_type == "firefox":
+            driver_class = firefox.webdriver.WebDriver
+            options = firefox.options.Options()
+        elif self._driver_type == "chrome":
+            driver_class = chrome.webdriver.WebDriver
+            options = chrome.options.Options()
+            options.add_argument(f"--window-size={self._window[0]},{self._window[1]}")
+        else:
+            raise Exception(f"Webdriver name ({self._driver_type}) not supported")
+        # Prepare args for the webdriver init
+
+        # Add additional configured options
+        for arg in current_app.config["WEBDRIVER_OPTION_ARGS"]:
+            options.add_argument(arg)
+
+        kwargs: Dict[Any, Any] = dict(options=options)
+        kwargs.update(current_app.config["WEBDRIVER_CONFIGURATION"])
+        logger.info("Init selenium driver")
+
+        return driver_class(**kwargs)
+
+    def auth(self, user: "User") -> WebDriver:
+        driver = self.create()
+        # Setting cookies requires doing a request first
+        driver.get(headless_url("/login/"))
+        return machine_auth_provider_factory.instance.authenticate_webdriver(
+            driver, user
+        )
+
+    @staticmethod
+    def destroy(driver: WebDriver, tries: int = 2) -> None:
+        """Destroy a driver"""
+        # This is some very flaky code in selenium. Hence the retries
+        # and catch-all exceptions
+        try:
+            retry_call(driver.close, tries=tries)
+        except Exception:  # pylint: disable=broad-except
+            pass
+        try:
+            driver.quit()
+        except Exception:  # pylint: disable=broad-except
+            pass
+
+    def get_screenshot(
+        self,
+        url: str,
+        element_name: str,
+        user: "User",
+        retries: int = SELENIUM_RETRIES,
+    ) -> Optional[bytes]:
+        driver = self.auth(user)
+        driver.set_window_size(*self._window)
+        driver.get(url)
+        img: Optional[bytes] = None
+        logger.debug("Sleeping for %i seconds", SELENIUM_HEADSTART)
```

Comment:
Does it make sense to create `SELENIUM_HEADSTART` has a config key also, `WEBDRIVER_HEADSTART`

Reply:
Probably in next PR :)